apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIrBuilder.kt

```diff
@@ -371,24 +382,41 @@ internal class BackendIrBuilder constructor(
     val fragmentInterfaces = this.fragments
         .mergeInterfaceFragmentsWithTheSameName()
 
+    val fragmentInterfaceRootSelectionKeys = fragmentInterfaces
+        .map { selectionKey + it.name }
+
     val fragmentImplementations = fragmentInterfaces
         .flattenFragments()
         .buildFragmentImplementations(
             parentName = this.typeName,
-            parentFields = this.fields,
+            // NOTE: required by new version that removes interfaces
+            parentFields = this.fields.addFieldSelectionKey(selectionKey),
```

Comment:
TBH not sure why this makes a new version of codegen that removes interfaces work but it breaks previous version. Will investigate in next PR.